include/onnxruntime/core/framework/provider_options.h

```diff
@@ -6,6 +6,19 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include "core/session/onnxruntime_c_api.h"
+
+/// <summary>
+/// This is CUDA provider specific but needs to live in a header that is build-flavor agnostic
+/// Options for the CUDA provider that are passed to SessionOptionsAppendExecutionProvider_CUDA
+/// </summary>
+typedef struct OrtCUDAProviderOptions {
```

Comment:
>OrtCUDAProviderOptions [](start = 15, length = 22)

perhaps there is some overloaded naming of "provider options". i'd suggest putting this in a separate header (cuda_provider_options.h?) so we can keep the includes more finely-grained, i.e. only including OrtCUDAProviderOptions where it's needed. if you can come up with a better name for the map<string, string> used to store provider option configs that'd be great too.

Reply:
Good suggestion. I moved the struct to cuda_provider_options.h, but it can't be moved to a cuda folder and has to remain in the same folder as provider_options.h because it needs to be available even in non-CUDA builds.

Reply:
I didn't quite understand the second part of the comment - the one about map<string, string>

Reply:
it's fine to leave it as ProviderOptions (the map<string, string>). we use "ProviderOptions" in the names of that typedef and the EP struct so i was just thinking it would be nice if the names were more different.